src/society/services/society.service.ts

```diff
@@ -61,6 +61,27 @@ export class SocietyService {
    */
   async addMemberToSociety(memDeets: addMemberSocietyDto): Promise<MemberOf> {
     const { sid, uid, ...restDeets } = memDeets;
+
+    // * This will add the member to the society in a single query but it's very hard to catch the proper error and return it to the user
```

Comment:
This commented out code is unsafe, and may allow for SQL injection. Please remove the commented out code.

You can try the transactions that as we discussed. Otherwise, we can ensure that it will fail if getMe() and getSocietyById() becomes invalid by the time the save is executed - no harm in that.

Reply:
I'll try transactions and revert back 👌

Reply:
check the below commit @chrisvrose :)